features/rule_creation_protocol/version_check.py

```diff
@@ -0,0 +1,29 @@
+import subprocess
+import re
+from .errors import ProtocolError
+
+def get_changed_feature_files(target_branch):
+    changed_files = subprocess.getoutput(f"git diff --name-only {target_branch}").split("\n")
```

Comment:
Please change these to `subprocess.run()` which is the "modern" way to do this, operating on a list. So that you don't need to worry about shell escaping. If `{file}` (next fn) contains a space, it will break. But if every arg is a separate str in a list passed to subprocess.run() it gets escaped automatically.